### PR TITLE
Fix field configuration view not updating when navigating between different fields in New Field drawer

### LINKED ISSUE #17128: Navigating between `Detail group` and `Raw Group` not updating configuration view
### Describe the Bug

When navigating between `Detail group` and `Raw Group`, in the `Create field` overlay panel, the content in the field configuration view doesn’t change.

### To Reproduce

Create a collection, `Create field`, click `Detail group` — configuration view opens —, click `Raw Group` — nothing happens —, click `Detail Group` — nothing happens —, click `Raw Group` — nothing happens —, click `Detail Group` — nothing happens —, click `Raw Group` — nothing happens —, click `Detail Group` — nothing happens —, click `Raw Group` — nothing happens —, click `Detail Group` — nothing happens —, click `Raw Group` — nothing happens —, click `Detail Group` — nothing happens  … ;)

### Errors Shown

_No response_

### What version of Directus are you using?

9.22.4

### What version of Node.js are you using?

Latest

### What database are you using?

MySQL

### What browser are you using?

Firefox

### How are you deploying Directus?

Docker

### PR DESCRIPTION
**Pull request by jay-p-b-7span**
## Description

Fixes #17128. I have mentioned the **explanation** of issue and **solution** below. Please have a look.

## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [ ] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

## Explanation of issue 
`extension-options` component containing extensionInfo variable which had `useExtension` composable value was not updating when different fields were selected. Due to which old object value was there and not getting updated in component. Its not reactive. To make it reactive, we have one solution i.e making argument `props.extension` computed. 

## Solution 
I made the `extensionInfo` object reactive by making `props.extension` to computed which then was updated. 
Screenshots attached. 
image
image

### CODE REVIEW

File: app/src/modules/settings/routes/data-model/field-detail/shared/extension-options.vue [hunk 1]
```diff
@@ -70,7 +70,10 @@ export default defineComponent({
 
 		const { collection, field } = storeToRefs(fieldDetailStore);
 
-		const extensionInfo = useExtension(props.type, props.extension);
+		const extensionInfo = useExtension(
+			props.type,
```
Comment (rijkvanzanten) [inline app/src/modules/settings/routes/data-model/field-detail/shared/extension-options.vue:74]: Won't this have the same issue if the type of extension changes?

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/modules/settings/routes/data-model/field-detail/shared/extension-options.vue [hunk 2]
```diff
@@ -70,7 +70,10 @@ export default defineComponent({
...
+			computed(() => props.extension)
```
Comment (rijkvanzanten) [inline app/src/modules/settings/routes/data-model/field-detail/shared/extension-options.vue:75]: Also lets use Vue's `toRefs` to make this reactive:

```js
const { extension } = toRefs(props):
```

Comment (jay-p-b-7span) [reply app/src/modules/settings/routes/data-model/field-detail/shared/extension-options.vue]: Done 👍

File: app/src/modules/settings/routes/data-model/field-detail/shared/extension-options.vue (hunk 1 above)
Comment (jay-p-b-7span) [reply app/src/modules/settings/routes/data-model/field-detail/shared/extension-options.vue]: According to current implementation, `extension-options` component receives static string from parent. Definitely in future, we can update `type` to reactive.

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**